gcetcbendorsement/cmd/root.go

```diff
@@ -96,15 +97,32 @@ func (b *bearerGetter) Get(url string) ([]byte, error) {
 	return io.ReadAll(resp.Body)
 }
 
+// The go-tdx-guest IsSupported signature is different from go-sev-guest. Add a compatibility layer.
+type tdxQuoteProvider struct{ p tdclient.QuoteProvider }
+
+func (qp *tdxQuoteProvider) IsSupported() bool { return qp.p.IsSupported() == nil }
+func (qp *tdxQuoteProvider) GetRawQuote(reportData [64]byte) ([]uint8, error) {
+	return qp.p.GetRawQuote(reportData)
+}
+
+func getProvider() (p extract.QuoteProvider) {
+	p, _ = client.GetQuoteProvider()
```

Comment:
It seems like the intention of this function to try many different quote provides and choose one that claims it's supported. I think it would be better to invert the if statement so that there's less indentation as more quote providers are added. Ideally it'd use a loop over an array of quote providers as well.

e.g.
```
let possible_providers = [client.GetQuoteProvider(), tdxQuoteProvider{p: tdclient.GetQuoteProvider()}];
possible_providers.find(|p| p.IsSupported());
```

Reply:
Creating the providers when you don't have to for constructing that list is a bit wasteful and leads to needing to do nil checking in the lambda, so for two entries I'd rather not.

Reply:
Well you have to create both for the TDX case anyway. So possibly it's an optimization for the SEV case. I have no idea how expensive creating these classes are. if they're expensive, maybe there's a cheaper check to run to quickly determine if they're supported?